base/MAPL_Resource.F90

```diff
@@ -0,0 +1,411 @@
+      end if
+
+      ! Do not print label more than once
+      if (.not. vector_contains_str(already_printed_labels, trim(label))) then
+         call already_printed_labels%push_back(trim(label))
+      else
+         return
+      end if
+
+      select type(val)
+      type is(integer(int32))
+         type_str = "'Integer*4 '"
+         type_format = '(i0.1)'
+         val_str = intrinsic_to_string(val, type_format)
+         if (present(default)) then
+            default_str = intrinsic_to_string(default, type_format)
+         end if
+      type is(integer(int64))
+         type_str = "'Integer*8 '"
+         type_format = '(i0.1)'
+         val_str = intrinsic_to_string(val, type_format)
+         if (present(default)) then
+            default_str = intrinsic_to_string(default, type_format)
+         end if
+      type is(real(real32))
+         type_str = "'Real*4 '"
+         type_format = '(f0.6)'
+         val_str = intrinsic_to_string(val, type_format)
+         if (present(default)) then
+            default_str = intrinsic_to_string(default, type_format)
+         end if
+      type is(real(real64))
+         type_str = "'Real*8 '"
+         type_format = '(f0.6)'
+         val_str = intrinsic_to_string(val, type_format)
+         if (present(default)) then
+            default_str = intrinsic_to_string(default, type_format)
+         end if
+      type is(logical)
+         type_str = "'Logical '"
+         type_format = '(l1)'
+         val_str = intrinsic_to_string(val, type_format)
+         if (present(default)) then
+            default_str = intrinsic_to_string(default, type_format)
+         end if
+      type is(character(len=*))
+         type_str = "'Character '"
+         val_str = trim(val)
+         if (present(default)) then
+            default_str = intrinsic_to_string(default, 'a')
+         end if
+         class default
+         _FAIL("Unsupported type")
+      end select
+
+      output_format = "(1x, " // type_str // ", 'Resource Parameter: '" // ", a"// ", a)"
+
+      ! printrc = 0 - Only print non-default values
+      ! printrc = 1 - Print all values
+      if (present(default)) then
+         if (trim(val_str) /= trim(default_str) .or. printrc == 1) then
+            print output_format, trim(label), trim(val_str)
+         end if
+      else
+         print output_format, trim(label), trim(val_str)
+      end if
+
+   end subroutine print_resource
+
+   logical function vector_contains_str(vector, string)
+      type(StringVector), intent(in) :: vector
+      character(len=*), intent(in) :: string
+      type(StringVectorIterator) :: iter
+
+      iter = vector%begin()
+
+      vector_contains_str = .false.
+
+      if (vector%size() /= 0) then
+         do while (iter /= vector%end())
+            if (trim(string) == iter%get()) then
+               vector_contains_str = .true.
+               return
+            end if
+            call iter%next()
+         end do
+      end if
```

Comment:
```suggestion
         do while (iter /= vector%end())
            if (trim(string) == iter%get()) then
               vector_contains_str = .true.
               return
            end if
            call iter%next()
         end do
```
The check on size is not necessary.   An empty vector starts the iterator equal to end.

I'd also like to update this to gFTL2 while we are looking at it.   Should just need to change the USE statement to reference `gftl2` and change `iter%get()` to `iter%of()`

Reply:
Done. Changed `use gFTL_StringVector` to `use gFTL2_StringVector`, eliminated size check, and changed `iter%get()` to `iter%of()`.